doc/fcn_2D_segm.txt

```diff
@@ -0,0 +1,210 @@
+    :align: center
+    :scale: 35%
+
+    **Figure 1** : Segmentation network (from FCN paper)
+
+TODO : reference de l'image
+
+**Fully Convolutional Networks** (FCNs) owe their name to their architecture, which is
+built only from locally connected layers, such as convolution, pooling and upsampling.
+Note that no dense layer is used in this kind of architecture. This reduces the number
+of parameters and computation time. To obtain a segmentation map (output), segmentation
+networks usually have 2 parts :
+
+*  Downsampling path : capture semantic/contextual information
+*  Upsampling path : recover spatial information
+
+The **downsampling path** is used to extract and interpret the context (*what*), while the
+**upsampling path** is used to enable precise localization (*where*). Furthermore, to fully
+recover the fine-grained spatial information lost in the pooling or downsampling layers, we
+often use skip connections.
+
+A skip connection is a connection that bypasses a least one layer. Here, it
+is often used to transfer local information by concatenating or summing feature
+maps from the downsampling path with feature maps from the upsampling path. Merging features
+from various resolution levels helps combining context information with spatial information.
+
+
+Data
+++++
+
+Polyps
+
+The polyps dataset can be found `here <https://drive.google.com/file/d/0B_60jvsCt1hhZWNfcW4wbHE5N3M/view>`__.
+In each of the training, validation and test directory, the input images are in the
+/images directory and the polyps mask (segmentation map) are in /masks2. The
+segmentation maps in the *masks2* directory indicate the presence or absence
+of polyps for each pixel. The other subdirectories (/masks3 and /masks4) are,
+respectively, for a segmentation task with 3 and 4 classes, but will not be
+presented here.
+
+
+Model
++++++
+
+There are variants of the FCN architecture, which mainly differ in the spatial precision of
+their output. For example, the figures below show the FCN-32, FCN-16 and FCN-8 variants. In the
+figures, convolutional layers are represented as vertical lines between pooling layers, which
+explicitely show the relative size of the feature maps.
+
+.. figure:: images/fcn.png
+    :align: center
+    :scale: 50%
+
+    **Figure 2** : FCN architecture (from FCN paper)
+
+**Difference between the 3 FCN variants**
+
+As shown below, these 3 different architectures differ in the stride of the last convolution,
+and the skip connections used to obtain the output segmentation maps. We will use the term
+*downsampling path* to refer to the network up to *pool5* layer and we will use the term
+*upsampling path* to refer to the network composed of all layers after *pool5*. It is worth
+noting that the 3 FCN architectures share the same downsampling path, but differ in their
+respective upmsapling paths.
+
+
+1. **FCN-32** : Directly produces the segmentation map from *pool5*, by using a
+transposed convolution layer with stride 32.
```

Comment:
What about conv6-7? Are they the transposed convolution you mention? But if there are two of them, which one has stride 32?
Or is it a transposed convolution after conv7?